local/compose/convergence.go

```diff
@@ -390,3 +390,31 @@ func (s *composeService) startService(ctx context.Context, project *types.Projec
 	}
 	return eg.Wait()
 }
+
+func (s *composeService) restartService(ctx context.Context, project *types.Project, service types.ServiceConfig, timeout *time.Duration) error {
+	containers, err := s.apiClient.ContainerList(ctx, moby.ContainerListOptions{
+		Filters: filters.NewArgs(
+			projectFilter(project.Name),
+			serviceFilter(service.Name),
+		),
+		All: true,
+	})
+	if err != nil {
+		return err
+	}
+	w := progress.ContextWriter(ctx)
+	eg, ctx := errgroup.WithContext(ctx)
+	for _, c := range containers {
```

Comment:
I wonder we need to run this in respect to depedencies order(especially with wait-condition: healthy
If we don't, then passing the project name would be enough and avoid requirement to get the original compose file

Reply:
Also, can we do a container restart on each container, or do we need to stop the whole project, and restart it ? 
not sure how this is done in `docker-compose`.

Reply:
@ndeloof And it does https://github.com/docker/compose-cli/pull/1415/files#diff-d6629ccbd8641e5ff4ab3b8cccf9b8a4c7921f765f3bc0bc9ba8ee47bc715c8eR27-R35
Actually `docker-compose` seems to get the containers in an unsorted way, see https://github.com/docker/compose/blob/5b7851f55b4ae9533f42eed282ffad7c442778f5/compose/project.py#L471 but as you stated, I preferred to visit it in dependencies order.

@gtardif `docker-compose` restart is an API call and not a combination of `stop` followed by `start`. Also doing so would lead to possible data synch problems, since the compose file could be different in between operations. Container ID's and other things could be differ too.

Reply:
Please don't invoke `ContainerList` for each service. Collect project containers once, then filter to get the service ones.
Command should not require a `project` but use the same logic you designed for `down` to rebuild model from project name (using labels)
We could later add labels for service depednencies so that we don't even need to retrieve and parse the compose file